src/borrow_tracker/tree_borrows/tree/tests.rs

```diff
@@ -75,151 +75,109 @@ fn protected_enforces_noalias() {
     }
 }
 
-/// Possible pairs of `AccessRelatedness` that can occur
-/// to two tags that are mutually foreign.
-/// If there are two pointers `x` and `y` and neither is a child of the other
-/// (which also means they are distinct), then any access is either
-/// - a child of `x` but foreign for `y` (`ChildX`)
-/// - a child of `y` but foreign for `x` (`ChildY`)
-/// - foreign for both (`Other`)
+/// Accessed pointer.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-enum AccessRel2MutuallyForeign {
-    ChildX,
-    ChildY,
+enum PtrSelector {
+    X,
+    Y,
     Other,
 }
 
-impl Exhaustive for AccessRel2MutuallyForeign {
+/// Relative position of `x` and `y`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum RelPosXY {
+    MutuallyForeign,
+    XChildY,
+}
```

Comment:
What bout the obvious 3rd case, `YChildX`?

Is it because X is definitely getting a retag, and hence cannot have children elsewhere? Please explain that in the doc comment. Really, this means you need to have a comment before all this that explains the set of patterns you are going to model, so that the reader has any chance of having the right context available.

Reply:
Hm no, y also always gets retagged. Which difference can XChildY make?

Reply:
`y` does not always get retagged, there's an `if state.y.state.initialized { ... }` that accounts for the fact that the creation of `y` might not involve a read to that location.

All cases are covered:
- `y` raw pointer to previously initialized location  -> `y` initialized retagged before `x`
- `y` raw pointer, out-of-bounds of latest retag -> `y` uninitialized not retagged
- `y` reference, retagge -> `y` initialized retagged either before or after `x`
- `y` reference, out-of-bounds of retag -> `y` uninitialized not retagged

Reply:
What I don't get is why RelPosXY is independent of y_retag.  If you already consider both MutuallyForeign and XChildY and should be enough, no?

> there's an if state.y.state.initialized { ... } that accounts for the fact that the creation of y might not involve a read to that location.

Reads are already part of the unknown code though, so this seems redundant.

Reply:
There are 3 relevant configurations:
- `XChildY`
- `MutuallyForeign` with `x` older than `y`
- `MutuallyForeign` with `y` older than `x`
the latter two are not trivially guaranteed to be equivalent, because some cases could be eliminated by a read-on-reborrow.

"Reads are already part of the unknown code" yes but the unknown code also includes sequences with no read at all that could appear in the test as false negatives, I think.

Reply:
So what about this: `OpaqueCode` also gets a `initial_y_retag: bool` field, to make it clear that we are treating this as part of the unknown code. Then it becomes part of `applicable_opaque_code_without_ub` to try things with and without the initial retag of `y`.

Do we have some argument for why doing the retag right at the start (and not in the middle) is sufficient?